IABTechLab/uid2-web-integrations
# Ans UI d2 3834 allow js sdk accept multiple init calls

What Changed:

- No more error on multiple init calls
- Different process whether init has already been called or if it is the first init call
- Params updated (baseUrl, identity if it expires after the current identity, cookiePath, cookieDomain, refreshRetryPeriod, useCookie
- Config updated if any params have changed or if the method of storing identity (cookie or local storage has changed)
- Way to store identity (cookie or local storage) updated if useCookie changes 
- initCallback is now a class, when init callback gets sent, it is added to an array of previous init callbacks
- Added new variable if aborted, abort no longer = init complete since we can have multiple init calls
- function `isInitialized` added so client knows if it has already been initialized, they might not want to do it again
- Tests added 
- per the ticket, logs added to help with debugging

Test Plan:
- Run the local cstg example from the web-integrations repo -> Run and debug -> Launch cstg (chrome).  Make sure local operator is running (go to `uid2-dev-workspace` -> run `docker compose up` -> and check `http://localhost:8080/ops/healthcheck` to make sure it is running 
- Change the `index.html` file in the cstg example to call init multiple times and change the parameters from SdkOptions each time.  Make sure everything still functions as normal in the example site (`baseUrl` has to be on localhost, but you can test that changing it to something else makes the example not work)
